src/main/webapp/app/exercises/programming/shared/code-editor/container/code-editor-container.component.html

```diff
@@ -5,9 +5,9 @@ <h4><ng-content select="[editorTitle]"></ng-content></h4>
         <div class="d-flex align-items-center ml-auto">
             <ng-content select="[editorToolbar]"></ng-content>
             <jhi-code-editor-actions
-                *ngIf="!isTutorAssessment"
+                *ngIf="!isTutorAssessment || commitState === CommitState.CONFLICT"
                 [buildable]="buildable"
-                [disableActions]="!editable"
+                [disableActions]="isTutorAssessment && !editable ? commitState !== CommitState.CONFLICT : !editable"
```

Comment:
Can this simplify to something like `!editable || isTutorAssessment && commitState !== CommitState.CONFLICT`?

Reply:
If !editable = false and isTutorAssessment = true and commitState != CommitState.CONFLICT then the return value for yours would be true and for Rene's it would be false. If I am not mistaken...

Reply:
Taking the time to resolve this I think it would be `!editable && (!isTutorAssessment || commitState !== CommitState.CONFLICT)` if we wanted to simplify this.

Reply:
Tbh, we can leave this as is when not sure about changing from a working version.

Reply:
@FrankeLukas  i like your last idea, I will implement it

```suggestion
                [disableActions]="!editable && (!isTutorAssessment || commitState !== CommitState.CONFLICT)"
```